cpp/src/arrow/ipc/writer.cc

```diff
@@ -1042,8 +1042,13 @@ class ARROW_EXPORT IpcFormatWriter : public RecordBatchWriter {
 
 class StreamBookKeeper {
  public:
-  explicit StreamBookKeeper(const IpcWriteOptions& options, io::OutputStream* sink)
+  StreamBookKeeper(const IpcWriteOptions& options, io::OutputStream* sink)
       : options_(options), sink_(sink), position_(-1) {}
```

Comment:
Do we need `owned_sink_(nullptr)` here?

Reply:
No, it's implicit (a `shared_ptr` is always initialized).